src/tiledb/cloud/bioimg/exportation.py

```diff
@@ -1,33 +1,78 @@
-from typing import Any, Dict, Mapping, Optional, Sequence, Tuple, Union
+import os
+from typing import Any, Dict, Iterator, Mapping, Optional, Sequence, Tuple, Union
 
 import tiledb
-from tiledb.cloud._common.utils import logger
-from tiledb.cloud.bioimg.helpers import batch
-from tiledb.cloud.bioimg.helpers import get_uris
-from tiledb.cloud.bioimg.helpers import scale_calc
+from tiledb.cloud import dag
+from tiledb.cloud.bioimg.helpers import get_logger_wrapper
+from tiledb.cloud.utilities._common import run_dag
 
 DEFAULT_RESOURCES = {"cpu": "8", "memory": "4Gi"}
 DEFAULT_IMG_NAME = "3.9-imaging-dev"
 DEFAULT_DAG_NAME = "bioimg-exportation"
+_RUNNING_PROFILES = ("client", "server")
+
+
+def build_io_uris_exportation(source: Sequence[str], output_dir: str, output_ext: str):
+    """Match input uri/s with output destinations
+
+    :param source: A sequence of paths or path to input
+    :param output_dir: A path to the output directory
+    """
+    vfs = tiledb.VFS()
+
+    def create_output_path(input_file, output_dir) -> str:
+        filename = os.path.splitext(os.path.basename(input_file))[0]
+        output_filename = filename + f".{output_ext}" if output_ext else filename
+        return os.path.join(output_dir, output_filename)
+
+    def iter_paths(source: Sequence) -> Iterator[Tuple]:
+        for uri in source:
+            if vfs.is_dir(uri) and tiledb.object_type(uri) != "group":
+                # Folder for exploration
+                contents = vfs.ls(uri)
+                yield from tuple(iter_paths(contents)[1:])
```

Comment:
Just noticed this—is this what you intend? Iterators don’t generally have indexing available. It seems likely that you would want:

```
yield from iter_paths(contents[1:])
```

Wrapping in `tuple` will also eagerly evaluate the entirety of the inner `iter_paths` call rather than picking from it one by one, which is fine but I would recommend adding a comment explaining *why*.

Reply:
After some debugging it seems that after the new release of TileDB the `vfs.ls` drops empty subfolders, so this `[1:]` is not needed anymore. I also changed the direct materialisation of the iterators results and now are fetched one at a time. There was no specific reasoning behind this after all just a design decision on my part.